source/dofs/dof_handler_policy.cc

```diff
@@ -1140,12 +1764,12 @@ namespace internal
                cell != dof_handler.end();
                ++cell)
             for (unsigned int q = 0; q < GeometryInfo<dim>::quads_per_cell; ++q)
-              if ((cell->quad(q)->user_flag_set() == false) &&
+              if ((cell->quad(q)->user_flag_set() == true) &&
                   (cell->quad(q)->n_active_fe_indices() == 2))
```

Comment:
Nice -- that just made this a bit cheaper than :-)

Vertices can't be queried this way. But it's cheap to iterate over them and test the condition you have, so I don't think much can go wrong. I was more worried about this here because you are looking at faces between ghost cells where you may not know whatever information you need. With the additional condition, this can't happen any more.

If you really wanted to do this for vertices as well, you'd have to loop over all ghost cells first and flag their vertices somehow. Then you'd loop over the locally owned cells, loop over their vertices, treat each, and if you're done with it remove the flag. There is no `cell->vertex(v)->set_user_flag()`, so you'll have to store these flags in a vector of bools, but I think you're already doing that.

Reply:
I don't think it matters for vertices, though.